arch-design/multicloud/BucketManagement.md

```diff
@@ -0,0 +1,369 @@
+
+
+**Author(s)**: [Pravin Ranjan](https://github.com/PravinRanjan10), [Vineela Pachchipulusu]()
+
+
+## Motivation and background
+
+In the cloud world, every cloud vendor provides storage space to users. Generally in terms of bucket and folder. A user can store the data/objects in a specific bucket or folder.
+
+SODA multi-cloud(gelato) is a project which manages multiple cloud vendors from one place. And also interfaces to use cloud services. So it is important for soda to provide bucket management support for various cloud vendors. So that user can create and manage buckets from one place(soda multi-cloud).
+
+## Goals
+The document provides the use case, design and implementation guidelines for supporting bucket management from SODA platform. It includes creation, deletion, object uploading, migration etc. in various cloud providers.
+
+### Non-Goals
+NA
+
+### Requirement Analysis
+
+##### Functional Requirements
+
+1. Create buckets in various cloud providers. Ex: Aws, gcp, azure
+
+2. Delete bucket from various cloud providers. Ex: Aws, gcp, azure
+
+3. Upload objects in bucket
+
+4. Set lifecycles rules on objects
+
+5. Migrate objects from one location to different locations
+
+
+##### Non Functional Requirements
+
+1. The system should be highly available. Multiple users can login to the system and create buckets, folders, upload objects etc.
+The design document for this can be referred from [here](https://github.com/sodafoundation/architecture-analysis/pull/96)
+
+2. The system should work with minimum latency.
+
+3. The system should be scalable and it should handle the high amount of users request.
+
+
+## Architecture Analysis
+
+The overall multi cloud architecture can be referred from [here](https://github.com/sodafoundation/architecture-analysis/blob/master/arch-design/multicloud/MultiCloud_Design.md)
+
+### Use case View
+
+For heterogeneous object management, bucket creation is important. Currently soda-multicloud supports creation of folders under cloud buckets as shown in Fig-1. So, soda-users must need to create cloud buckets manually and provide a name, AK/SK to soda for folder creation. As shown in fig-1, step-1 is manual which is performed by the user, whereas step-2,3,4,.. are supported by soda-multi-cloud.
+
+To avoid manual intervention, step-1 should also be part of soda-multicloud as shown in Fig-2. This way users need not to do anything manually on the cloud vendors console.
+
+Once the buckets are created, users should perform all the operations supported by soda-multicloud. For example: Uploading objects, setting lifecycle rules, migration etc.  
+
+
+![](resources/multi-cloud-fig1.png)
+
+
+![](resources/multi-cloud-fig2.png)
+
+
+
+#### API's(Create/Delete) to support bucket management:
+
+To support bucket managament, below is the list of API's with detailed description about parameters for various cloud vendors are below:
+
+
+#### AWS
+
+##### Rules for bucket naming:
```

Comment:
Are these rules generic and standard enough considering all cloud vendors support?

Reply:
These rules for just refernces. I have updated the docyment.